src/index.js

```diff
@@ -94,19 +95,34 @@ export default class AuthnWidget {
   }
 
   init() {
-    try {
-      if (!this.flowId) {
-        throw new Error(AuthnWidget.FLOW_ID_REQUIRED_MSG);
+    if (window.opener !== null) {
+      // for external authentication in popup, refresh parent window
```

Comment:
Can you elaborate on why this is needed in the init method? It seems to me that this method wouldn't be called again after the widget is created for a page

Reply:
for the popup window case, the widget is initialized again after successful login inside the popup window. I wanted to stop this from happening and instead trigger another `GET` inside parent window for the new authentication state.

Reply:
I see, so it is expected that when the popup is closed the parent windows is refreshed?

Reply:
yeah, the popup triggers parent page refresh then closes itself.

Reply:
Does this work if the widget is embedded in another page and it opened a popup? I'm also thinking the window.opener [does not work in IE](https://developer.mozilla.org/en-US/docs/Web/API/Window/opener), has that been tested out yet?

Reply:
I believe it should work, but I'll test it out to confirm since I can't recall which implementation I had shown during that previous meeting. As for IE support, I'll need to confirm as well. I know `window.opener` works fine for IE, just not sure whether `location` is defined. Is IE support something we want? I just assumed it wasn't since `fetch` is used, but it [doesn't seem to be supported by IE](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API)?

Reply:
Yeah we claim support for IE11 In the readme we have:
> The widget works with all major browsers, including Chrome, Firefox, IE11, and MS Edge.

Fetch works in IE since we pull in whatwg-fetch as a dependency to allow for IE11 support.